src/neuroglancer/datasource/computed/backend.ts

```diff
@@ -0,0 +1,508 @@
+    vec3.min(inBoxUpper, inBoxUpper, originSource.spec.upperVoxelBound);
+    const gridUpper =
+        vec3.floor(vec3.create(), vec3.divide(vec3.create(), inBoxUpper, originChunkSize));
+
+    const gridPosition = vec3.create();
+    for (let z = gridLower[2]; z <= gridUpper[2]; ++z) {
+      for (let y = gridLower[1]; y <= gridUpper[1]; ++y) {
+        for (let x = gridLower[0]; x <= gridUpper[0]; ++x) {
+          gridPosition.set([x, y, z]);
+          const key = gridPositionKey(gridPosition);
+          this.originGridPositions_.set(key, vec3.copy(vec3.create(), gridPosition));
+        }
+      }
+    }
+
+    for (const chunkGridPosition of this.originGridPositions_.values()) {
+      const chunk = originSource.getChunk(chunkGridPosition);
+      chunk.registerListener(this);
+      this.stateChanged(chunk);
+    }
+  }
+}
+
+@registerSharedObject() export class ComputedVolumeChunkSource extends
+(WithParameters(VolumeChunkSource, ComputedVolumeChunkSourceParameters)) {
+  originSource: VolumeChunkSource;
+  private activeComputations_ = new Map<String, ComputedVolumeChunk>();
+  private frontendChunkRequests_ =
+      new Map<String, Map<String, (array: ArrayType, error: any) => void>>();
+  public computation: VolumeComputationBackend;
+
+  constructor(rpc: RPC, options: any) {
+    super(rpc, options);
+    this.originSource = this.rpc!.getRef<VolumeChunkSource>(this.parameters.sourceRef);
+    this.computation = this.rpc!.getRef<VolumeComputationBackend>(this.parameters.computationRef);
+    this.registerDisposer(this.chunkManager);
+    this.registerDisposer(this.chunkManager.recomputeChunkPrioritiesLate.add(() => {
+      this.updateChunkPriorities();
+    }));
+  }
+
+  updateChunkPriorities() {
+    const priorityMap = new Map<String, {priority: number, tier: number, pos: vec3}>();
+
+    for (const outputChunk of this.activeComputations_.values()) {
+      for (const gridPosition of outputChunk.getOverlappingOriginGridPositions()) {
+        const gridString = gridPosition.toLocaleString();
+        if (!priorityMap.has(gridString)) {
+          priorityMap.set(
+              gridString,
+              {priority: outputChunk.priority, tier: outputChunk.priorityTier, pos: gridPosition});
+        } else {
+          const currentPriority = priorityMap.get(gridString)!;
+          currentPriority.priority = Math.max(currentPriority.priority, outputChunk.priority);
+          currentPriority.tier = Math.min(currentPriority.tier, outputChunk.priorityTier);
+        }
+      }
+    }
+
+    for (const priorityInfo of priorityMap.values()) {
+      const chunk = this.originSource.getChunk(priorityInfo.pos);
+      if (priorityInfo.tier !== ChunkPriorityTier.RECENT) {
+        this.chunkManager.requestChunk(chunk, priorityInfo.tier, priorityInfo.priority);
+      }
+    }
+  }
+
+  unregisterChunk(chunk: ComputedVolumeChunk) {
+    const key = chunk.key!;
+    this.activeComputations_.delete(key);
+  }
+
+  requestChunkData(
```

Comment:
You could simplify this logic by using registerPromiseRPC / promiseInvoke (see e.g. datasource/nifti/backend.ts and datasource/nifti/frontend.ts) --- it would eliminate the need for you to separately manage frontendChunkRequests, cancellation, etc.

Reply:
As I mentioned in a side conversation, this turned out to be much more difficult to implement than I'd hoped, but I might give it another go.

Reply:
Done.